src/init/starship.zsh

```diff
@@ -62,19 +62,17 @@ starship_zle-keymap-select() {
     zle reset-prompt
 }
 
-## Check for existing keymap-select widget.
 # zle-keymap-select is a special widget so it'll be "user:fnName" or nothing. Let's get fnName only.
 __starship_preserved_zle_keymap_select=${widgets[zle-keymap-select]#user:}
```

Comment:
Please also update this to set the value if `__starship_preserved_zle_keymap_select` was not unset, to avoid inadvertently replacing the actual contents with `starship_zle-keymap-select-wrapped`. This could also be used to avoid installing the wrapper if the value is `starship_zle-keymap-select`.

Reply:
Thanks for the review!  

Do you mean changing the code like this?  

```zsh
if [[ -z $__starship_preserved_zle_keymap_select ]]; then
    __starship_preserved_zle_keymap_select=${widgets[zle-keymap-select]#user:}
fi
```

I actually considered this too. However, I think it may cause an issue.  

For example, consider this `.zshrc`:  

```zsh
bindkey -v
function myfoo() {
        echo "\nfoo\n"
}

function mybar() {
        echo "\nbar\n"
}
zle -N zle-keymap-select myfoo
eval "$(~/github-project/starship/target/debug/starship init zsh)"
```

If the user later wants to change `zle-keymap-select` to `mybar`, then `source .zshrc`:

```zsh
--- zle -N zle-keymap-select myfoo
+++ zle -N zle-keymap-select mybar
```

This would cause the user’s changes to disappear.  

Maybe I misunderstood your suggestion; please pointer it out if that’s the case.

Reply:
I think something like below could account for that. I would also prefer not defining a wrapper function unless needed:

```sh
__starship_preserved_zle_keymap_select_capture=${widgets[zle-keymap-select]#user:}
if [[ -n $__starship_preserved_zle_keymap_select_capture ]] && [[ $__starship_preserved_zle_keymap_select_capture == "starship_zle-keymap-select-wrapped" ||  $__starship_preserved_zle_keymap_select_capture == "starship_zle-keymap-select" ]]; then
     __starship_preserved_zle_keymap_select=$__starship_preserved_zle_keymap_select_
else

if [[ -z $__starship_preserved_zle_keymap_select ]]; then
    zle -N zle-keymap-select starship_zle-keymap-select;
else
# ....
```

Reply:
This sounds like the [temporary solution](https://github.com/starship/starship/issues/3418#issuecomment-2477375663) I used before.

I also agree not to define a wrapper function if it's unneeded.